scene/2d/tile_map.cpp

```diff
@@ -1501,12 +1501,16 @@ void TileMap::draw_tile(RID p_canvas_item, const Vector2 &p_position, const Ref<
 		} else {
 			real_t speed = atlas_source->get_tile_animation_speed(p_atlas_coords);
 			real_t animation_duration = atlas_source->get_tile_animation_total_duration(p_atlas_coords) / speed;
+			bool sync = atlas_source->get_tile_animation_sync(p_atlas_coords);
+			int frames_count = atlas_source->get_tile_animation_frames_count(p_atlas_coords);
+			int start_frame = sync ? 0 : rand() % frames_count;
 			real_t time = 0.0;
-			for (int frame = 0; frame < atlas_source->get_tile_animation_frames_count(p_atlas_coords); frame++) {
-				real_t frame_duration = atlas_source->get_tile_animation_frame_duration(p_atlas_coords, frame) / speed;
+			for (int frame = 0; frame < frames_count; frame++) {
+				int current_frame = (frame + start_frame) % frames_count;
+				real_t frame_duration = atlas_source->get_tile_animation_frame_duration(p_atlas_coords, current_frame) / speed;
 				RenderingServer::get_singleton()->canvas_item_add_animation_slice(p_canvas_item, animation_duration, time, time + frame_duration, 0.0);
 
-				Rect2i source_rect = atlas_source->get_runtime_tile_texture_region(p_atlas_coords, frame);
+				Rect2i source_rect = atlas_source->get_runtime_tile_texture_region(p_atlas_coords, current_frame);
 				tex->draw_rect_region(p_canvas_item, dest_rect, source_rect, modulate, transpose, p_tile_set->is_uv_clipping());
 
 				time += frame_duration;
```

Comment:
Currently what's being randomized is only the starting frame, which I don't think is the way to go. Like e.g. for an animation with 2 frames with durations 1 and 100 respectively it would still pick each frame with ~50% probabiility. I suggest changing it to be time based.

For this you could do (more or less) something like `real_t animation_offset = sync ? 0.0 : randf() * animation_duration;` and keep the loop as it was before, changing only the last argument of `canvas_item_add_animation_slice` to be such `animation_offset`.

Besides that, such `animation_offset` would need to be stored per cell, instead of choosing a new random offset each time such cell is being redrawn (by calling `draw_tile`).

Reply:
I was originally intending to use the probability values that are already stored and set per tile, like in TileMapEditorTilesPlugin::_pick_random_tile? Perhaps that would be better? Otherwise, probability exists but is meaningless for animation tiles.

Reply:
Sorry for my mistake, not available for all animation tiles, because cannot select all, and already used by Place Random Tile.